packages/contracts-bedrock/test/L1/OptimismPortal.t.sol

```diff
@@ -1127,41 +1225,139 @@ contract OptimismPortalWithMockERC20_Test is OptimismPortal_FinalizeWithdrawal_T
         token = new MockERC20("Test", "TST", 18);
     }
 
-    /// @dev Tests that `depositERC20Transaction` succeeds.
-    function testFuzz_depositERC20Transaction_succeeds(uint256 _amount) external {
+    /// @dev Tests that `depositERC20Transaction` succeeds when msg.sender == tx.origin.
+    function testFuzz_depositERC20Transaction_senderIsOrigin_succeeds(
+        address _to,
+        uint256 _mint,
+        uint256 _value,
+        uint64 _gasLimit,
+        bool _isCreation,
+        bytes memory _data
+    )
+        external
+    {
+        if (_isCreation) {
+            vm.assume(_to == address(0));
+        }
+        vm.assume(_data.length <= 120_000);
+        vm.assume(_gasLimit >= optimismPortal.minimumGasLimit(uint64(_data.length)));
+        ResourceMetering.ResourceConfig memory rcfg = systemConfig.resourceConfig();
+        vm.assume(int256(uint256(_gasLimit)) <= int256(uint256(rcfg.maxResourceLimit)));
+
         // Mint the token to the contract and approve the token for the portal
-        token.mint(address(this), _amount);
-        token.approve(address(optimismPortal), _amount);
+        token.mint(address(this), _mint);
+        token.approve(address(optimismPortal), _mint);
 
         // Mock the gas paying token to be the ERC20 token
         vm.mockCall(address(systemConfig), abi.encodeWithSignature("gasPayingToken()"), abi.encode(address(token), 18));
 
+        bytes memory opaqueData = abi.encodePacked(_mint, _value, _gasLimit, _isCreation, _data);
+
         vm.expectEmit(address(optimismPortal));
         emit TransactionDeposited(
-            AddressAliasHelper.applyL1ToL2Alias(address(this)),
-            address(0),
-            uint256(0),
-            abi.encodePacked(
-                _amount, // mint
-                uint256(0), // value
-                optimismPortal.minimumGasLimit(0), // gasLimit
-                false, // isCreation,
-                "" // data
-            )
+            address(this), // from
+            _to,
+            uint256(0), // DEPOSIT_VERSION
+            opaqueData
         );
 
+        // Ensure that msg.sender == tx.origin
+        vm.prank(address(this), address(this));
+
         // Deposit the token into the portal
-        optimismPortal.depositERC20Transaction(address(0), _amount, 0, optimismPortal.minimumGasLimit(0), false, "");
+        optimismPortal.depositERC20Transaction(_to, _mint, _value, _gasLimit, _isCreation, _data);
+
+        // Assert final balance equals the deposited amount
+        assertEq(token.balanceOf(address(optimismPortal)), _mint);
+        assertEq(optimismPortal.balance(), _mint);
+    }
+
+    /// @dev Tests that `depositERC20Transaction` succeeds when msg.sender != tx.origin.
+    function testFuzz_depositERC20Transaction_senderNotOrigin_succeeds(
```

Comment:
Similar to above, you can deduplicate code by using an internal function with the logic and have the 2 fuzz tests call the internal function. The internal function can take an extra arg that says which prank/expect emit should be used